[BE-3062]: Added danger docs (#430)
appcircleio/appcircle-docusaurus: docs/workflows/common-workflow-steps/danger.md

---
title: Danger
description: Streamline your code review process with Danger. Automate checks and enforce code standards pre-merge to maintain high-quality software.
tags: [code review, quality assurance, automaiton, development tools, appcircle, mobile ci/cd]
sidebar_position: 4
---

import Screenshot from '@site/src/components/Screenshot';

# Danger

> [Danger](https://danger.systems/ruby/) is a CI tool that allows you to automate code review. It not only helps code reviewers but also the developers who send pull requests. The reviewer spends less time doing chores and more time efficiently reviewing the code.

Reviewer: ```suggestion
[Danger](https://danger.systems/ruby/) automates code reviews through CI tools, aiding both code reviewers and developers who submit pull requests. It reduces the time reviewers spend on routine tasks, allowing more efficient code evaluation.
```

> PRs opened can be easily automated with Appcircle's Danger integration.

Reviewer: ```suggestion
You can easily automate opened PRs with Appcircle's Danger integration.
```

For detailed information on the benefits that Danger, please refer to the following blog post:

https://appcircle.io/blog/danger-in-ci-automate-your-mobile-code-reviews

:::warning

This tool does not support AzureDevOps. Therefore, if your repository is hosted on AzureDevOps, this tool will not function. Please use the [**Azure Bot for Swiftlint**](/workflows/ios-specific-workflow-steps/azure-bot-for-swiftlint)(for iOS) and [**Azure Bot for Detekt Report**](/workflows/android-specific-workflow-steps/azure-bot-for-detekt-report)(for Android)  components instead.

:::

### Prerequisites

The workflow steps that need to be executed before running the **Danger** step vary depending on the platform and are listed below:

| Prerequisite Workflow Step                      | Description                                     |
|-------------------------------------------------|-------------------------------------------------|
| [Git Clone](/workflows/common-workflow-steps/git-clone) | The repository needs to be cloned to begin the badge-adding process. After this step, the variable `AC_REPOSITORY_DIR` will be set. |

:::caution

> Please note that this component works in synchronization with the [**Appcircle Triggers**](/build/build-process-management/build-manually-or-with-triggers/#managing-triggers-for-builds). If a trigger is not set up, the pipeline will not be triggered when a PR is opened, and therefore, Danger will not function.

Reviewer: ```suggestion
Note that this component synchronizes with the [**Appcircle Triggers**](/build/build-process-management/build-manually-or-with-triggers/#managing-triggers-for-builds). Without a setup trigger, opening a PR will not trigger the pipeline, and Danger will not function.
```
